src/pages/HomePage.js

```diff
@@ -1,47 +1,53 @@
-import React, { useContext } from "react";
+import React, { useContext, useEffect } from "react";
 
 import Container from "react-bootstrap/Container";
 import Col from "react-bootstrap/Col";
 import Row from "react-bootstrap/Row";
 import { Context } from "../contex";
+import getCityWeather from "../apis/weather-api";
 
 const HomePage = () => {
-  const { weather } = useContext(Context);
+  const { weather, setWeather } = useContext(Context);
+
+  useEffect(() => {
+    const fetchWeather = async () => {
+      const city = "London";
+      const cityWeather = await getCityWeather(city);
+      setWeather(cityWeather);
+      console.log(weather);
+    };
+
+    fetchWeather();
```

Comment:
Let's use `await` in front of `fetchWeather` function calling and get the result to set weather.

Reply:
We are awaiting above with 'await.' Do we need to use it

Reply:
```
  const { weather, setWeather } = useContext(Context);
  const { city } = useState("London");

  useEffect(() => {
    setWeather(getCityWeather(city));
  }, [city, setWeather]);
  ```